### PR TITLE
Build changes for COIN

### PR DESCRIPTION
**Pull request by bennibbelink**
This makes appropriate build changes in #1613 from the upstream repo to follow our CI best practices

### CODE REVIEW

**Comment by bennibbelink** [comment]
Hmm I didn't quite think this through.  This PR has the appropriate build changes but without the appropriate source code changes in #1613 it will fail the cbc unit tests (not skip them anymore since `--allow-milps` is enabled)

File: tests/prog_translator_tests.cc [hunk 1]
```diff
@@ -243,7 +243,7 @@ TEST(ProgTranslatorTests, translation) {
   EXPECT_NO_THROW(SolveProg(iface));
   const double* soln = iface->getColSolution();
   const double* check = checkface.getColSolution();
-  array_double_eq(soln, check, narcs + nfaux, "soln");
+  array_double_near(soln, check, narcs + nfaux, 0.00000000001, "soln");
```
Comment (bennibbelink) [inline tests/prog_translator_tests.cc:246]: This margin was chosen arbitrarily

Comment (gonuke) [reply tests/prog_translator_tests.cc:246]: I think there is a cyclus definition of small somewhere - maybe `eps`?  Not sure if/how it's available in testing

File: CHANGELOG.rst [hunk 2]
```diff
@@ -37,6 +37,7 @@ Since last release
 * fix sell_policy that was offering bids when capacity was inbetween 0 and the
   quantize, bids that one was not able to fullfill and caused cyclus to crash. (#1552)
 * Deprecation warnings involving <boost/detail/sp_typeinfo.hpp> (#1611)
+* Segmentation faults when calling Cbc (#1614)
```
Comment (gonuke) [inline CHANGELOG.rst:40]:
```suggestion
* Resolves segmentation faults when calling Cbc (#1614)
```

File: tests/prog_translator_tests.cc (hunk 1 above)
Comment (gonuke) [reply tests/prog_translator_tests.cc:246]: `cy_eps`

Comment (bennibbelink) [reply tests/prog_translator_tests.cc]: Had to include `cyc_limits.h`, now we pass in cy_eps as the margin

File: CHANGELOG.rst (hunk 2 above)
Comment (bennibbelink) [reply CHANGELOG.rst]: Addressed this, as well as the previous fix in `CHANGELOG.rst` that was similarly worded

**Review by gonuke — APPROVED** [review]
Thanks for more cleanup @bennibbelink

**Pull request merged**